libs/CCAppCommon/include/ccApplicationBase.h

```diff
@@ -23,16 +23,33 @@
 
 //! Mimic Qt's qApp for easy access to the application instance
 #define ccApp (static_cast<ccApplicationBase *>( QCoreApplication::instance() ))
-
+//! ccApplicationBase
+/*!
+ * ccApplicationBase provides a QApplication class with additional Constructor param isCommandLine and version
+ * The class also initialises openGL before instantiating the application class
+ * \see QApplication
+ */
 class CCAPPCOMMON_LIB_API ccApplicationBase : public QApplication
 {
 public:
 	//! This must be called before instantiating the application class so it
 	//! can setup OpenGL first.
 	static void	initOpenGL();
-	
+
+    //! ccApplicationBase Constructor
+	/*!
+	 * @param argc argc must be greater than zero, this is set by QT
+	 * @param argv this contains the command line values, see QApplication, again this is set by QT
+	 * @param isCommandLine boolean to define the application as either a Commandline or GUI. If true its a commandline application
```

Comment:
Could we condense the documentation a little? Generally it's not useful to recall the type of the variable or class (already visible in the doxygen output). Something like:
'Whether the application is to be used in command line mode (true) or GUI mode (false).'

Reply:
ok will change. thank you